src/battleship/Battleship.java

```diff
@@ -63,88 +79,115 @@ public int colNum(String coordinate) {
-            }
+    public void playerTwo() {
+        playerOneHidden.printField();
+        System.out.println("---------------------");
+        playerTwoField.printField();
+    }
 
-            int row = rowNum(coordinate.get());
-            int col = colNum(coordinate.get());
+    public void start() {
 
-            if (row < 0 || col < 0) {
-                System.out.println("Invalid location. Try again.");
-                continue;
-            }
+        setPlayerBoard(playerOneShips, playerOneField);
+        System.out.println("Press Enter and pass the move to another player");
+        String input = scanner.nextLine();
 
-            try {
-                if (board.isShip(row, col)) {
-                    hiddenBoard.setIndex(row, col, 'X');
-                    board.setIndex(row, col, 'X');
-                    hiddenBoard.printField();
-                    for (Ship ship : ships) {
-                        board.isSunken(ship);
-                    }
-                    boolean sank = false;
-                    for (Ship type : ships) {
-                        if (type.isSunken()) {
-                            sank = true;
-                            ships.remove(type);
-                            break;
-                        }
-                    }
-                    if (sank) {
-                        System.out.println("You sank a ship! Specify a new target:");
-                    } else {
-                        System.out.println("You hit a ship! Try again:");
-                    }
-                    oCounter--;
-                } else if (board.isEmpty(row, col)) {
-                    hiddenBoard.setIndex(row, col, 'M');
-                    board.setIndex(row, col, 'M');
-                    hiddenBoard.printField();
-                    System.out.println("You missed. Try again:");
-                } else if (board.isHit(row, col)) {
-                    hiddenBoard.printField();
-                    System.out.println("You already hit this one. Try again:");
-                } else if (board.isMiss(row, col)) {
-                    hiddenBoard.printField();
-                    System.out.println("You already missed this one. Try again:");
-                }
-            } catch (IllegalArgumentException | ArrayIndexOutOfBoundsException e) {
-                System.out.println("Error! You entered the wrong coordinates! Try again:");
-            }
+        if (input.isEmpty()) {
+            System.out.println("Player 2, place your ships to the game field");
+            setPlayerBoard(playerTwoShips, playerTwoField);
+        }
+        System.out.println("Press Enter and pass the move to another player");
+        String secondInput = scanner.nextLine();
+
+        if (secondInput.isEmpty()) {
+            playerOneField.totalShips();
+            playerTwoField.totalShips();
+            playerOneShipsOnField = playerOneField.getO();
+            playerTwoShipsOnField = playerTwoField.getO();
```

Comment:
You should be able to avoid having to scan the board to find out what happened.. You should store and update the totals as the board is updated.